### DISCUSSION TITLE
Should private repositories hide HTTPS URLs?

### DISCUSSION THREAD

**Discussion by KonradHoeffner** [community/community#52164] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

1. private repositories require authentication to clone
2. HTTPS authentication uses a password
3. password authentication is not allowed anymore
4. private repositories give an HTTPS URL under main page -> Code -> Local -> Clone -> HTTPS

1.+2.+3.+4. leads to confusing behavior for new GitHub users, in that they want to clone a repository and enter "git clone https://github.com/someorg/private-repo.git" and get denied.
As far as I understand it, it does not make sense to show an essentially unusable HTTPS URL on private repos.

**Comment by airtower-luna** [comment, marked as answer]
You can't use your login password for HTTPS auth, but you can use a [personal access token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) with the right scopes (**repo**, plus **workflow** if you want to be able to push changes to Actions workflows) instead of the password. This is not only for private repositories, but also for pushing to public ones over HTTPS.

**Comment by KonradHoeffner** [comment]
Ah, I didn't know that. Thanks for the explanation!

**Comment by airtower-luna** [comment]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:
